### PR TITLE
Sub ordinate view

### PR DESCRIPTION
**Pull request by Asna-A**

### CODE REVIEW

File: src/services/WolfDen.API/Controllers/Attendence/Attendence.cs [hunk 1]
```diff
@@ -90,6 +92,13 @@ public async Task<IActionResult> CheckAttendanceClose([FromQuery] CheckAttendanc
             return Ok(isClosed);
         }
 
+        [HttpGet("get-subordinates")]
```
Comment (Narayana221) [inline src/services/WolfDen.API/Controllers/Attendence/Attendence.cs:95]: [HttpGet("subordinates")] is better

File: src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/SubOrdinatesQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{
+    public class SubOrdinatesQueryHandler(WolfDenContext context) : IRequestHandler<SubOrdinatesQuery,SubOrdinateDTO>
+    {
+        private readonly WolfDenContext _context = context;
+
+        public async Task<SubOrdinateDTO> Handle(SubOrdinatesQuery request, CancellationToken cancellationToken)
+        {
+            TeamHeirarchyService service = new(_context);
+            Employee employee = await _context.Employees.FindAsync(request.EmployeeId, cancellationToken);
+            if (employee == null)
```
Comment (Narayana221) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/SubOrdinatesQueryHandler.cs:18]: employee is null

File: src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/ManagerEmailFinder.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            _context = context;
+        }
+        public async Task<List<string>> FindManagerEmailsAsync(int? managerId, CancellationToken cancellationToken)
+        {
+            List<string> managerEmails = new List<string>();
+            if (managerId == null) 
+                return managerEmails;
+
+            var manager = await _context.Employees
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/ManagerEmailFinder.cs:20]: remove var

File: src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/ManagerEmailFinder.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                .Where(m => m.Id == managerId)
+                .Select(m => new { m.Email, m.ManagerId })
+                .FirstOrDefaultAsync(cancellationToken);
+
+            if (manager != null)
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/ManagerEmailFinder.cs:25]: manager is not null

File: src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/ManagerEmailFinder.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        private readonly WolfDenContext _context;
+
+        public ManagerEmailFinder(WolfDenContext context)
+        {
+            _context = context;
+        }
+        public async Task<List<string>> FindManagerEmailsAsync(int? managerId, CancellationToken cancellationToken)
+        {
+            List<string> managerEmails = new List<string>();
+            if (managerId == null)
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/ManagerEmailFinder.cs:17]: is null

File: src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/SendEmailCommandHandler.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        public SendEmailCommandHandler(WolfDenContext context, IConfiguration configuration,ManagerEmailFinder emailFinder)
+        {
+            _context = context;
+            _senderEmail = configuration["BrevoApi:SenderEmail"];
+            _senderName = configuration["BrevoApi:SenderName"];
+            _emailFinder = emailFinder;
+        }
+        public async Task<string> Handle(SendEmailCommand request, CancellationToken cancellationToken)
+        {
+            var employee = await _context.Employees
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/SendEmailCommandHandler.cs:26]: remove var

File: src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/SendEmailCommandHandler.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                    Name = senderName
+                };
+
+                List<SendSmtpEmailTo> toList = receiverEmails.Select(email => new SendSmtpEmailTo(email)).ToList();
+                List<SendSmtpEmailCc> ccList = ccEmails?.Select(email => new SendSmtpEmailCc(email)).ToList() ?? new List<SendSmtpEmailCc>();
+
+                SendSmtpEmail sendSmtpEmail = new SendSmtpEmail(sender, toList, null, ccList, null, message, subject);
+                CreateSmtpEmail result = apiInstance.SendTransacEmail(sendSmtpEmail);
+
+                Console.WriteLine($"Email sent successfully: {result.ToJson()}");
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Commands/Attendence/SendEmail/SendEmailCommandHandler.cs:58]: remove

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyDetails/DailyDetailsQueryHandler.cs [hunk 8]
```diff
@@ -7,17 +7,17 @@
 
 namespace WolfDen.Application.Requests.Queries.Attendence.DailyStatus
 {
-    public class DailyDetailsQueryHandler :IRequestHandler<DailyDetails,DailyAttendanceDTO>
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyDetails/DailyDetailsQueryHandler.cs:10]: format

File: src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace WolfDen.Application.Requests.Services
+{
+    public class TeamHeirarchyService(WolfDenContext context)
+    {
+        private readonly WolfDenContext _context = context;
+        public async Task<List<SubOrdinateDTO>> GetSubordinates(int managerId)
+        {
+            List<SubOrdinateDTO> result = new List<SubOrdinateDTO>();
+            List<Employee> employees = await _context.Employees.Where(x => x.ManagerId == managerId && x.IsActive == true).Include(x=>x.Department).Include(x=>x.Designation).ToListAsync();
```
Comment (Narayana221) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs:14]: x.IsActive , no need to equate it to true

File: src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/SubOrdinatesQueryHandler.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        public async Task<SubOrdinateDTO> Handle(SubOrdinatesQuery request, CancellationToken cancellationToken)
+        {
+            TeamHeirarchyService service = new(_context);
+            Employee employee = await _context.Employees.FindAsync(request.EmployeeId, cancellationToken);
+            if (employee is null) 
+            {
+                return null;
+            }
+            SubOrdinateDTO subOrdinate = new SubOrdinateDTO();
+            if (employee.IsActive == false)
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/SubOrdinatesQueryHandler.cs:23]: There's no need for two separate if conditions that return null; both conditions can be combined into a single if statement.

File: src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            foreach (Employee employee in employees)
+            {
+                SubOrdinateDTO employeeDto = new SubOrdinateDTO();
+                employeeDto.Id = employee.Id;
+                employeeDto.EmployeeCode = employee.EmployeeCode;
+                employeeDto.Name = employee.FirstName + " " + employee.LastName;
+                employeeDto.Email = employee.Email;
+                employeeDto.Photo = employee.Photo;
+                if(employee.Department is not null)
+                employeeDto.Department = employee.Department.Name;
```
Comment (Narayana221) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs:24]: employeeDto.Department = employee.Department?.Name;

Comment (Narayana221) [reply src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs:24]: can omit null check

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyDetails/PdfService.cs [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+                                          .Padding(10).AlignCenter().Text("Device Name").Bold().FontSize(12);
+                                    header.Cell().Border(1).BorderColor(Colors.Black)
+                                          .Background(Colors.Grey.Lighten3)
+                                          .Padding(10).AlignCenter().Text("Direction").Bold().FontSize(12);
+                                });
+                                 foreach (var attendenceLog in dailyStatusDTO.DailyLog)
+                                 {
+                                     string direction = "in";
+                                     table.Cell().Border(1).Padding(10).Text(attendenceLog.Time.ToString("HH:mm")).FontSize(12);
+
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyDetails/PdfService.cs:119]: remove unwanted space and format

File: src/services/WolfDen.API/Controllers/Attendence/Attendence.cs [hunk 13]
```diff
@@ -28,16 +30,16 @@ public Attendance(IMediator mediator, PdfService pdfService)
...
         }
 
         [HttpGet("daily-attendance-pdf")]
-        public async Task<IResult> GeneratePdf([FromQuery] DailyDetailsPdf DailyDetailspdf, CancellationToken cancellationToken)
+        public async Task<IResult> GeneratePdf([FromQuery]DailyDetailsQuery DailyDetails, CancellationToken cancellationToken)
         {
-            DailyAttendanceDTO attendenceList = await _mediator.Send(DailyDetailspdf, cancellationToken);
+            DailyAttendanceDTO attendenceList = await _mediator.Send(DailyDetails, cancellationToken);
             var document = _pdfService.CreateDocument(attendenceList);
             var pdf = document.GeneratePdf();
```
Comment (muhammedashikCQ) [inline src/services/WolfDen.API/Controllers/Attendence/Attendence.cs:44]: remove var

File: src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs (hunk 9 above)
Comment (Asna-A) [reply src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs:14]: but it is a nullable field

Comment (Narayana221) [reply src/services/WolfDen.Application/Requests/Queries/Attendence/SubOrdinates/TeamHeirarchyService.cs:14]: okay understood

**Review by Narayana221 — APPROVED** [review]

**Pull request merged**